src/conversions/rust_decimal.rs

```diff
@@ -64,8 +64,10 @@ impl FromPyObject<'_> for Decimal {
         if let Ok(val) = obj.extract() {
             Ok(Decimal::new(val, 0))
         } else {
-            Decimal::from_str(&obj.str()?.to_cow()?)
-                .map_err(|e| PyValueError::new_err(e.to_string()))
+            Decimal::from_str(&obj.str()?.to_cow()?).or_else(|_| {
```

Comment:
Could you pull out the `obj.str()?.to_cow()?` part so that it is evaluated only once?

Reply:
Done!

Reply:
Please keep the let bindings limited to the `else` branch though. Otherwise, there is unnecessary work in the fast path for extracting an `i64`.

Reply:
Oh sorry about that, I thought i had done it on the `else` branch, my bad. Fixed , and thanks for the feedback!